src/app/complex/complex-results/complex-navigator/table-structure/table-interactor-column/table-interactor-column.component.ts

```diff
@@ -234,74 +236,27 @@ export class TableInteractorColumnComponent implements OnChanges {
     return enrichedComplex;
   }
 
-  public classifyInteractorsByOrganism() {
-    this.enrichedInteractors.sort((a, b) => {
-      if (b.interactor.organismName === a.interactor.organismName) {
-        return b.timesAppearing - a.timesAppearing;
-      } else {
-        const organismBTimesAppearing = this._timesAppearingByOrganism.get(b.interactor.organismName);
-        const organismATimesAppearing = this._timesAppearingByOrganism.get(a.interactor.organismName);
-        if (organismBTimesAppearing === organismATimesAppearing) {
-          return b.interactor.organismName.localeCompare(a.interactor.organismName);
-        } else {
-          return organismBTimesAppearing - organismATimesAppearing;
-        }
-      }
-    });
-    this.rangeOfInteractorOrganism();
-  }
+  private compareFn = (a: EnrichedInteractor, b: EnrichedInteractor) =>
+    -(b.indexAppearing - a.indexAppearing) || // First by order of appearance (staircase effect)
+    -(b.timesAppearing - a.timesAppearing); // Then by reversed occurrence, in order to minimize edge length
 
-  public classifyInteractorsByType() {
-    this.enrichedInteractors.sort((a, b) => {
-      if (b.interactor.interactorType === a.interactor.interactorType) {
-        return b.timesAppearing - a.timesAppearing;
-      } else {
-        const typeBTimesAppearing = this._timesAppearingByType.get(b.interactor.interactorType);
```

Comment:
This logic using `_timesAppearingByType` and `_timesAppearingByOrganism` was to display first the types or organism with the most complexes. Now we're back to sorting them by name.

The changes in sort by occurrence makes sense, also the change to sort interactors with the same type or organism; but this other change we should not do.

Reply:
If I search for "*", and then group by type; currently the proteins go first and then the small molecules (at the bottom of the table). https://complex-portal.github.io/complex-portal-view/complex/search?query=*#view_navigator

<img width="1182" alt="Screenshot 2024-09-05 at 08 41 04" src="https://github.com/user-attachments/assets/bb45c5cd-483c-486b-840e-5199a102a5d1">

With these changes the small molecules are displayed first (at the top of the table) and then the proteins, as we simply sort the types alphabetically.

<img width="1081" alt="Screenshot 2024-09-05 at 08 41 15" src="https://github.com/user-attachments/assets/16c5bd0a-3693-429e-a5fb-9ae4e06e7146">